x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageOwnApiKeyClusterPrivilege.java

```diff
@@ -75,6 +76,10 @@ protected boolean extendedCheck(String action, TransportRequest request, Authent
                         invalidateApiKeyRequest.getUserName(), invalidateApiKeyRequest.getRealmName(),
                         invalidateApiKeyRequest.ownedByAuthenticatedUser()));
                 }
+            } else if (request instanceof QueryApiKeyRequest) {
+                final QueryApiKeyRequest queryApiKeyRequest = (QueryApiKeyRequest) request;
+                queryApiKeyRequest.setFilterForCurrentUser();
+                return true;
```

Comment:
As discussed, this is the simple way of applying calling user's security context. It works but is probably not a long term solution. I am open to alternatives.

Reply:
I assume your example is about the interceptor approach. We don't have such an interceptor for cluster actions, but I assume it would work similarly to the interceptors for index requests. The interface is

```java
void intercept(
  RequestInfo requestInfo, 
  AuthorizationEngine authorizationEngine, 
  AuthorizationInfo authorizationInfo,
  ActionListener<Void> listener);
```

Are you concerning about that, inside this method, we need to only rely on the fact that both `AuthorizationEngine` and `AuthorizationInfo` are interfaces and we need do it in a way that does not make `RBACEngine` and our `Role` special for at least in this method? It's challenging even with the "separate action name" approach. None of the existing methods fit this purpose really well. We can sorta fit it into `AuthorizationEngine#authorizeClusterAction`. But it would be better if we can add a new method to `AuthorizationEngine`. I think we'll need high bandwidth conversation on this.

Reply:
I came up with a [new approach](https://github.com/elastic/elasticsearch/pull/75335/commits/786daad61b6a54859083feb3c1f9613911fbccef) inspired by your comment. The idea is to authorize the action a second time if it fails for the first time and it is the QueryApiKey action. I think this approach is promising because:
* It is simple. A few lines change in `AuthorizationService`
* In theory works with arbitrary `AuthorizationEngine` and `AuthorizationInfo`
* Does not need a separate action name and works kinda smoothly with the `manage_own_api_key` privilege.

It has the overhead of a second authorization for the QueryApiKey action. But I think it is acceptable because it authorizing on name is pretty fast and it does not affect any other actions.